test/Microsoft.AspNet.Mvc.Core.Test/Filters/FormatFilterTest.cs

```diff
@@ -0,0 +1,378 @@
+            Assert.Equal(1, objectResult.ContentTypes.Count);
+            AssertMediaTypesEqual(mediaType, objectResult.ContentTypes[0]);
+        }
+
+        [Theory]
+        [InlineData("foo", FormatSource.RouteData, "application/foo")]
+        [InlineData("foo", FormatSource.QueryData, "application/foo")]
+        public void FormatFilter_ContextContainsNonExistingFormat(
+            string format,
+            FormatSource place,
+            string contentType)
+        {
+            // Arrange  
+            var mediaType = MediaTypeHeaderValue.Parse(contentType);
+            var resourceExecutingContext = CreateResourceExecutingContext(new IFilter[] { }, format, place);          
+            var filter = new FormatFilterAttribute();
+
+            // Act
+            filter.OnResourceExecuting(resourceExecutingContext);
+
+            // Assert
+            var actionResult = resourceExecutingContext.Result;
+            Assert.IsType<HttpNotFoundResult>(actionResult);
+        }
+
+        [Fact]
+        public void FormatFilter_ContextDoesntContainFormat()
+        {
+            // Arrange
+            var resourceExecutingContext = CreateResourceExecutingContext(new IFilter[] { });
+            var filter = new FormatFilterAttribute();
+
+            // Act
+            filter.OnResourceExecuting(resourceExecutingContext);
+
+            // Assert
+            Assert.Null(resourceExecutingContext.Result);
+        }
+
+        [Theory]
+        [InlineData("json", FormatSource.RouteData, "application/json")]
+        [InlineData("json", FormatSource.QueryData, "application/json")]
+        public void FormatFilter_ContextContainsFormat_ContainsProducesFilter_Matching(
+            string format, 
+            FormatSource place, 
+            string contentType)
+        {
+            // Arrange
+            var produces = new ProducesAttribute(contentType, new string[] { "application/foo", "text/bar" });
+            var context = CreateResourceExecutingContext(new IFilter[] { produces }, format, place);
+            var filter = new FormatFilterAttribute();
+
+            // Act
+            filter.OnResourceExecuting(context);
+
+            // Assert            
+            Assert.Null(context.Result);
+        }
+
+        [Fact]
+        public void FormatFilter_ContextContainsFormat_ContainsProducesFilter_WildCardMatching()
+        {
+            // Arrange
+            var produces = new ProducesAttribute(
+                "application/baz", 
+                new string[] { "application/foo", "text/bar" });
+            var context = CreateResourceExecutingContext(new IFilter[] { produces }, "star", FormatSource.RouteData);
+            var options = context.HttpContext.RequestServices.GetService<IOptions<MvcOptions>>();
+            options.Options.FormatterMappings.SetFormatMapping("star", MediaTypeHeaderValue.Parse("application/*"));
```

Comment:
this is the opposite of what I'd expect. I'd expect that the `application/*` goes into produces. 

I good example of this in practice would be an api that returns images in a few formats. You'd have `[Produces("image/*")]` on the action, and use a format mapping to get the exact type you want.